gnosis/gp-v2-dune
# Dune Query Templating

This is a first approach at generating Dune queries via a templating language that should allow us to reuse the chunks of code appearing in several tables throughout our Dashboards. 

This first example demonstrates how we can build the [Quick Trade History Table](https://dune.xyz/queries/229637) via templates.

## Test Plan
1. Run the new script (if this is your first time using this repo, some venv should be initialized with requirements installed)
```sh
python query_archive/generate.py
```
and find the generated results file `trade_history.gen.sql` written to `query_archive/.`

2. Create a new Dune query at dune.xyz with the contents of these results 
3. Run the query and compare with corresponding [Quick Trade History Table](https://dune.xyz/queries/229637)
